src/main/java/com/aws/greengrass/deployment/ThingGroupHelper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment;
+
+import com.aws.greengrass.componentmanager.ComponentServiceHelper;
+import com.aws.greengrass.deployment.exceptions.NonRetryableDeploymentTaskFailureException;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import com.aws.greengrass.util.RetryUtils;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.greengrassv2data.model.ListThingGroupsForCoreDeviceRequest;
+import software.amazon.awssdk.services.greengrassv2data.model.ListThingGroupsForCoreDeviceResponse;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.inject.Inject;
+
+public class ThingGroupHelper {
+    protected static final Logger logger = LogManager.getLogger(ComponentServiceHelper.class);
+    public static final String THING_GROUP_RESOURCE_TYPE = "thinggroup";
```

Comment:
Why is this prefix needed?

Reply:
The API returns thing group name which corresponds to resourseId in the arn, Nucleus stores the mapping of group to components using the resource path which is made of resourceType + / + resourceId. This way nucleus can differentiate deployments to thing groups and deployments to thing (device deployment).